File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        int count = 0, compareValue = 0;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == compareValue){
                count++;
                compareValue = nums[i];
            }else {
                compareValue = nums[i];
            }
        }

//1,2,3,3
//i = 0: nums[i] = 1 compareValue = 0 | false => compareValue = 1
//i = 1: nums[i] = 2 compareValue = 1 | false => compareValue = 2
//i = 2: nums[i] = 3 compareValue = 3 | false =
        if(count >= 1){
            return true;
        } 
        return false;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        int n = nums.size();
        int i = 0, left = 1, right = n - 1;
        sort(nums.begin(), nums.end());
        vector<int> sub;
        vector<vector<int>> ans;
        while(i < n){
            int sum = nums[i] + nums[left] + nums[right];
            if(sum < 0){
                left++
            } else if(sum > 0){
                right--;
            } else {
                sub.push_back(sum);
                left++;
                right--;
            }
            ans.push_back(sub);
        }
        return ans;
    }
};